explorer/interpreter/interpreter.cpp

```diff
@@ -2231,19 +2232,23 @@ auto Interpreter::StepStmt() -> ErrorOr<Success> {
         if (definition.init().kind() == ExpressionKind::CallExpression &&
             definition.init().expression_category() ==
                 ExpressionCategory::Initializing) {
-          // TODO: Handle forwarding allocation to nested initializing
-          // expressions.
-          // Allocate storage for initializing expression.
-          const auto allocation_id =
-              heap_.AllocateValue(arena_->New<UninitializedValue>(
-                  &definition.init().static_type()));
-          act.set_location_created(allocation_id);
-          RuntimeScope scope(&heap_);
-          scope.BindLifetimeToScope(Address(allocation_id));
-          todo_.MergeScope(std::move(scope));
+          if (!(definition.is_returned() && act.location_received())) {
```

Comment:
Join the ifs?

i.e., if (kind && expression_category && !(is_returned && location_received))